File: f77/build03/digsim.f
C
      Program DigSim
C
C     DigSim provides a FORTRAN architecture required to simulate
C     continuous systems described by sets of simultaneous first-order
C     differential equations.
C
C     Variables:
C        End_Of_Run - Flag set by user to stop the simulation [Logical]
C
C     Include common block variables and declarations
C
      Include '../driver/global.inc'
      Include '../driver/sysvars.inc'
      Include '../driver/deriv.inc'
C
C     Exceptions to default type
C
      Logical End_Of_Run,
     &        Execute_Run
C
C     Open the input file
C
      Open(Unit = 99, File = 'input.dat', Status = 'OLD')
C
C     Global driver initialization
C
      Call DigSim_Initialization
      Call Driver_Default_Data
C
C     Get data for first run
C
      Call Input_Data(Execute_Run)
C
C     If data is available, begin loop to execute simulation runs.
C
      Do While (Execute_Run)
C
C     Setup the driver for this run
C
         End_Of_Run = .FALSE.
         Call Zero_Global_Common
         Call Driver_Default_Data
         Call Default_Data
         Call Setup_Driver_For_Run
         Ndes_Real = 0
         Call Initialization
C
C     States at current time are known; calculate state derivatives at
C     current time
C
         Call Differential_Equations
C
C     Advance the simulation states until termination.
C
         Do While (.not. End_Of_Run)
C
C     Print the current values
C
            Call Print_Data
C
C     Advance states to new time
C
            Call Advance_States
C
C     Calculate state derivatives at new time
C
            Call Differential_Equations
C
C     Check the termination conditions
C
            Call Termination_Conditions(End_Of_Run)
         End Do
C
C     Setup for next run
C
         Call Input_Data(Execute_Run)
C
C     End of run execution loop
C
      End Do
C
      Stop
      End

File: compcert/build03/digsim.c
#include "../driver/global.h"
#include "../driver/sysvars.h"
#include "../driver/deriv.h"
#include "../driver/drvrdata.h"
#include "../driver/input.h"
#include "../driver/zerocomm.h"
#include "../driver/advance.h"
#include "../driver/digtypes.h"
#include "def_data.h"
#include "diffeq.h"
#include "drvrinit.h"
#include "init.h"
#include "prn_data.h"
#include "setup.h"
#include "term.h"

int main() {
  // DigSim provides a FORTRAN architecture required to simulate
  // continuous systems described by sets of simultaneous first-order
  // differential equations.
  //
  // Variables:
  //     End_Of_Run - Flag set by user to stop the simulation [Logical]
  BOOL end_of_run , execute_run;
  // Open the input file
  FILE *input;
  input = fopen("input.dat","r");
  digsim_initialization(); // Global driver initialization
  driver_default_data();
  execute_run = input_data(input); // Get data for first run
  // If data is available, begin loop to execute simulation runs.
  while (execute_run) {
    // Setup the driver for this run
    end_of_run = FALSE;
    zero_global_common();
    driver_default_data();
    default_data();
    setup_driver_for_run();
    ndes_real = -1;
    initialization();
    // States at current time are known; calculate state derivatives at current
    // time.
    differential_equations();
    // Advance the simulation states until termination.
    while (end_of_run != TRUE) {
      // Print the current values
      print_data();
      // Advance states to new time
      advance_states();
      // Calculate state derivatives at new time
      differential_equations();
      // Check the termination conditions
      end_of_run = termination_conditions();
    }
    // Setup for next run
    execute_run =input_data(input);
    // End of run execution loop
  }
  fclose(input);
}
